### DISCUSSION TITLE
Walrus Library

### DISCUSSION THREAD

**Discussion by klren0312** [creatorsand/.github#107] [Ideas]
### 一句话 IDEA：
实现去中心化书籍共享平台，用户可以上传书籍，和发表书评

### 代码仓库
https://github.com/klren0312/walrus-library

### demo地址
https://walrus-library.walrus.site/

### 我的角色：
开发

### 社群昵称：
ZCDC

### 个人简介：
前端工程师，其他也会点

### 产品是给谁用的？
想要共享书籍和分享书评的人

#### 痛点：

- 中心化平台依赖中心化的服务器存储数据，容易受到封禁、攻击或宕机的影响
- 用户的访问记录可能被平台存储，导致隐私泄露

#### 解决方案：

- 去中心化存储：利用walrus存储电子书，避免单点故障和数据丢失
- 用户可以通过去中心化身份匿名访问资源

#### 1. 用户可以上传书籍（需要mint创作者nft）
> 书籍文件和封面存储于walrus

领取创作者nft

https://github.com/user-attachments/assets/2288df1d-ef1f-437c-8a91-8c0171fd556c

分享书籍

https://github.com/user-attachments/assets/b99f52f2-f6b1-4b5e-89a6-720b1afee9c9

#### 2. 可以预览书籍和下载

https://github.com/user-attachments/assets/32157823-1dc1-4d9b-9c0c-390522a33905

#### 3. 可以发表书评和查看书评
> 书评markdown编辑器支持上传图片到walrus，书评markdown文本存于walrus

https://github.com/user-attachments/assets/edf1fd66-8096-4ab5-a83a-4f9e05935274

#### 4. 捐赠功能
> 存储服务由平台付费，用户可以捐赠指定sui，可在nft中记录捐赠值，方便后续激励

![image](https://github.com/user-attachments/assets/b2a51176-a79a-43ee-8a2f-633c0650e261)

**Comment by v1xingyue** [comment]
ai打分

[a comment by an automated account is not shown]
